app/src/main/java/com/github/epfl/meili/forum/FirebasePostService.kt

```diff
@@ -0,0 +1,79 @@
+package com.github.epfl.meili.forum
+
+import android.util.Log
+import com.github.epfl.meili.forum.Post.Companion.toPost
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.tasks.await
+import java.util.*
+
+class FirebasePostService() : PostService() {
+
+    companion object {
+        val DEFAULT_DB = { FirebaseFirestore.getInstance() }
+
+        // Change this for dependency injection
+        var dbProvider: () -> FirebaseFirestore = DEFAULT_DB
+    }
+
+    private val TAG = "FirebasePostService"
+    private var db: FirebaseFirestore = dbProvider()
+
+    init {
+        // Listen to changes in post collection and notify observers when it has changed
+        db.collection("posts").addSnapshotListener { snapshot, e ->
+            // Handle errors
+            if (e != null) {
+                Log.w(TAG, "Listen failed.", e)
+                return@addSnapshotListener
+            }
+
```

Comment:
In this line transform snapshot to list of ForumPosts and save it to service variable called posts: ArrayList<ForumPost>

Reply:
@Bwananah could you please explain the purpose of this listener? If I understood correctly you are using "get" on the collection reference whenever you want to get documents, so the listener isn't doing anything, right?

If you do what @MarcelTorne said above, you could actually get rid of the asynchronous "get" methods. That is how I implemented FirestoreReviewService in #53

Reply:
@Bwananah could you please explain the purpose of this listener? If I understood correctly you are using "get" on the collection reference whenever you want to get documents, so the listener isn't doing anything, right?

If you do what @MarcelTorne said above, you could actually get rid of the asynchronous "get" methods. That is how I implemented FirestoreReviewService in #53

Reply:
I changed that yesterday thanks to Marcel. See the code below.